### PR TITLE
ring_hash_lb: simulation test

### PR DESCRIPTION
**Pull request by danielhochman**
Was using this test to check distribution of keys among instances in the ring hash load balancer.

### CODE REVIEW

File: test/common/upstream/ring_hash_lb_test.cc [hunk 1]
```diff
@@ -142,5 +142,45 @@ TEST_F(RingHashLoadBalancerTest, UnevenHosts) {
...
+/**
+ * This test is for simulation only and should not be run as part of unit tests.
+ */
+class DISABLED_RingHashLoadBalancerTest : public RingHashLoadBalancerTest {
+public:
+  DISABLED_RingHashLoadBalancerTest() : RingHashLoadBalancerTest(){};
+};
+
+TEST_F(DISABLED_RingHashLoadBalancerTest, DetermineSpread) {
+
```
Comment (mattklein123) [inline test/common/upstream/ring_hash_lb_test.cc]: nit: del newline

File: test/common/upstream/ring_hash_lb_test.cc [hunk 2]
```diff
@@ -142,5 +142,45 @@ TEST_F(RingHashLoadBalancerTest, UnevenHosts) {
   }
 }
 
+/**
+ * This test is for simulation only and should not be run as part of unit tests.
```
Comment (mattklein123) [inline test/common/upstream/ring_hash_lb_test.cc]: For someone that wants to come along and maybe play with this, can you add a bit more docs in terms of what things to set, what the output looks like, etc.

Comment (danielhochman) [reply test/common/upstream/ring_hash_lb_test.cc]: updated

File: test/common/upstream/ring_hash_lb_test.cc [hunk 3]
```diff
@@ -142,5 +142,61 @@ TEST_F(RingHashLoadBalancerTest, UnevenHosts) {
...
+ * more time to build the ring, and slightly slower lookup when determining the backend for a
+ * key.
+ */
+class DISABLED_RingHashLoadBalancerTest : public RingHashLoadBalancerTest {
+public:
+  DISABLED_RingHashLoadBalancerTest() : RingHashLoadBalancerTest(){};
+};
+
+TEST_F(DISABLED_RingHashLoadBalancerTest, DetermineSpread) {
+  uint64_t num_hosts = 100;
```
Comment (mattklein123) [inline test/common/upstream/ring_hash_lb_test.cc]: nit: const for these vars.

**Pull request merged**